Sources/Sentry/SentryRequest.m

```diff
@@ -32,6 +33,34 @@ - (instancetype)init
     return serializedData;
 }
 
+- (void)setHeaders:(nullable NSDictionary<NSString *, NSString *> *)headers
+{
+    _headers = [SentryRequest sanitizedHeaders:headers];
```

Comment:
`h`: If I'm not mistaken, this is a change in behavior. This will also apply if a user explicitly sets headers on `SentryRequest`. In that case, we should keep the headers, see

> Note that if a user explicitly sets a request on the scope, nothing is stripped from that request. The above rules only apply to integrations that come with the SDK. -  https://develop.sentry.dev/sdk/data-handling/#sensitive-data